### PR TITLE
Moved Saml code to S.IM.Tokens.

### PR DESCRIPTION
**Pull request by brentschmaltz**
Removed a dependency.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24872697%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23issuecomment-74795939%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24873738%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23issuecomment-74796091%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24904109%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24904150%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23issuecomment-74872574%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20ce0e599914eabd69bd731326f20c2ea4ff05c746%20src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24872697%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20do%20we%20need%20to%20add%20.xproj%3F%20Is%20that%20going%20to%20override%20the%20.kproj%20file%3F%22%2C%20%22created_at%22%3A%20%222015-02-18T01%3A32%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%20the%20.xproj%20overwrites%20the%20.kproj.%20%20This%20is%20already%20in%20the%20project.%20VS%20automatically%20adds%20this%20line%20every%20time%2C%20so%20I%20decided%20to%20check%20in%20the%20change%20so%20I%20don%27t%20get%20nagged%20all%20the%20time.%22%2C%20%22created_at%22%3A%20%222015-02-18T14%3A31%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj%3AL1-35%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23issuecomment-74795939%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20do%20we%20have%20the%20SamlSecurityToken.cs%20and%20Saml2SecurityToken.cs%20classes%3F%20Why%20they%20are%20excluded%20in%20original%20project.json%20but%20not%20in%20S.IM.Tokens%20project.json%3F%22%2C%20%22created_at%22%3A%20%222015-02-18T01%3A48%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-02-18T01%3A50%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22thanks.%22%2C%20%22created_at%22%3A%20%222015-02-18T14%3A32%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ce0e599914eabd69bd731326f20c2ea4ff05c746%20src/System.IdentityModel.Tokens/project.json%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96%23discussion_r24873738%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20do%20the%20SamlSecurityToken.cs%20and%20Saml2SecurityToken.cs%20classes%20live%3F%20Why%20they%20are%20excluded%20in%20original%20project.json%20but%20not%20in%20S.IM.Tokens%20project.json%3F%22%2C%20%22created_at%22%3A%20%222015-02-18T01%3A49%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/803796%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tushargupta51%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20moved%20them%20under%20s.im.tokens%5C%5Csaml%20similar%20to%20jwt.%22%2C%20%22created_at%22%3A%20%222015-02-18T14%3A32%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens/project.json%3AL3-21%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull ce0e599914eabd69bd731326f20c2ea4ff05c746 src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj 1'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96#discussion_r24872697'>File: src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj:L1-35</a></b>
- <a href='https://github.com/tushargupta51'></a> Why do we need to add .xproj? Is that going to override the .kproj file?
- <a href='https://github.com/brentschmaltz'></a> yes the .xproj overwrites the .kproj.  This is already in the project. VS automatically adds this line every time, so I decided to check in the change so I don't get nagged all the time.
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96#issuecomment-74795939'>General Comment</a></b>
- <a href='https://github.com/tushargupta51'></a> Where do we have the SamlSecurityToken.cs and Saml2SecurityToken.cs classes? Why they are excluded in original project.json but not in S.IM.Tokens project.json?
- <a href='https://github.com/tushargupta51'></a> :+1:
- <a href='https://github.com/brentschmaltz'></a> thanks.
- [ ] <a href='#crh-comment-Pull ce0e599914eabd69bd731326f20c2ea4ff05c746 src/System.IdentityModel.Tokens/project.json 9'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96#discussion_r24873738'>File: src/System.IdentityModel.Tokens/project.json:L3-21</a></b>
- <a href='https://github.com/tushargupta51'></a> Where do the SamlSecurityToken.cs and Saml2SecurityToken.cs classes live? Why they are excluded in original project.json but not in S.IM.Tokens project.json?
- <a href='https://github.com/brentschmaltz'></a> I moved them under s.im.tokens\saml similar to jwt.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/96'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj]: Why do we need to add .xproj? Is that going to override the .kproj file?

**Comment by tushargupta51** [comment]
Where do we have the SamlSecurityToken.cs and Saml2SecurityToken.cs classes? Why they are excluded in original project.json but not in S.IM.Tokens project.json?

File: src/System.IdentityModel.Tokens/project.json [hunk 2]
```diff
@@ -3,12 +3,18 @@
     "dependencies": {
         "Newtonsoft.Json": "6.0.6"
     },
+    "exclude": [
+        "Saml\\SamlSecurityTokenHandler.cs",
+        "Saml\\Saml2SecurityTokenHandler.cs",
+        "Saml\\SamlConstants.cs",
+        "Saml\\SecurityKeyResolver.cs",
+        "Tokens\\SecurityTokenHandlerCollectionExtensions.cs"
```
Comment (tushargupta51) [inline src/System.IdentityModel.Tokens/project.json]: Where do the SamlSecurityToken.cs and Saml2SecurityToken.cs classes live? Why they are excluded in original project.json but not in S.IM.Tokens project.json?

**Comment by tushargupta51** [comment]
:+1:

File: src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj (hunk 1 above)
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj]: yes the .xproj overwrites the .kproj.  This is already in the project. VS automatically adds this line every time, so I decided to check in the change so I don't get nagged all the time.

File: src/System.IdentityModel.Tokens/project.json (hunk 2 above)
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens/project.json]: I moved them under s.im.tokens\saml similar to jwt.

**Comment by brentschmaltz** [comment]
thanks.

File: src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj (hunk 1 above)
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj]: :+1:<a href='#crh-update-none'></a>

File: src/System.IdentityModel.Tokens/project.json (hunk 2 above)
Comment (tushargupta51) [inline src/System.IdentityModel.Tokens/project.json]: :+1:<a href='#crh-update-complete'></a>

**Comment by tushargupta51** [comment]
:+1:<a href='#crh-update-none'></a>

File: src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj (hunk 1 above)
Comment (tushargupta51) [inline src/Microsoft.IdentityModel.Protocol.Extensions/Microsoft.IdentityModel.Protocols.xproj]: We have .kproj in other branches. we had .xproj earlier for a while in one of the VS update but now it's .kproj again.

**Pull request merged**